packages/core/assistant/src/session/session.ts

```diff
@@ -202,8 +175,72 @@ export class AiSession {
   }
 }
 
+/**
+ * Build a combined toolkit from the blueprint tools and the provided toolkit.
+ */
+const createToolkit = <Tools extends AiTool.Any>({
+  blueprints = [],
+  toolkit,
+}: Pick<SessionRunParams<Tools>, 'blueprints' | 'toolkit'>) =>
+  Effect.gen(function* () {
+    const blueprintToolkit = yield* ToolResolverService.resolveToolkit(blueprints.flatMap(({ tools }) => tools));
+    const blueprintToolkitHandler: Context.Context<AiTool.ToHandler<AiTool.Any>> = yield* blueprintToolkit.toContext(
+      ToolExecutionService.handlersFor(blueprintToolkit),
+    );
+
+    return yield* AiToolkit.merge(...[toolkit, blueprintToolkit].filter(isNotFalsy)).pipe(
+      Effect.provide(blueprintToolkitHandler),
+    ) as Effect.Effect<AiToolkit.ToHandler<any>, never, AiTool.ToHandler<Tools>>;
+  });
+
+/**
+ * Formats the system prompt.
+ */
+// TODO(burdon): Move to AiPreprocessor.
+const formatSystemPrompt = ({
+  system,
+  blueprints,
+  objects = [],
+}: Pick<SessionRunParams<any>, 'system' | 'blueprints' | 'objects'>) =>
+  Effect.gen(function* () {
+    const b = yield* pipe(
```

Comment:
`const b` ?? you're going against your core principles

Reply:
ha ha. no logging mistake.